Use Operations in CreateCertificateCommand

DechengMa opened the pull request:
We have implemented the `CreateCertificateCommand`, this PR is to apply Operations (our new way of fetching data) to this command 

# 📝 Summary of Changes

Changes proposed in this pull request:

- Add function `createCertificate` to `AppStoreConnectService`
- Create struct CreateCertificateOperation and CreateCertificateOptions
- Apply these to `CreateCertificateCommand`

# 🧐🗒 Reviewer Notes

## 💁 Example

Usage: 

`asc certificates create [--api-issuer <uuid>] [--api-key-id <keyid>] [--csv] [--json] [--table] [--yaml] <certificate-type> <csr-file>`

## 🔨 How To Test

`swift run asc certificates create MAC_INSTALLER_DISTRIBUTION filePath`

adamjcampbell commented:
@DechengMa this PR needs to be updated as I converted the command to make use of `await`

DechengMa commented:
> @DechengMa this PR needs to be updated as I converted the command to make use of `await`

Hi @adamjcampbell , I have rebased master and added .await() to the command

adamjcampbell commented on Tests/appstoreconnect-cliTests/Operations/CreateCertificateOperationTests.swift:
```diff
@@ -0,0 +1,85 @@
+        let operation = CreateCertificateOperation(options: options)
+
+        let result = Result {
+            operation.execute(with: dependencies)
```
```suggestion
            try operation.execute(with: dependencies).await()
```

adamjcampbell commented on Tests/appstoreconnect-cliTests/Operations/CreateCertificateOperationTests.swift:
```diff
@@ -0,0 +1,85 @@
+        }
+
+        switch result {
+            case .success(let publisher):
+                _ = publisher.sink(
+                receiveCompletion: { _ in },
+                receiveValue: { certificate in
+                    XCTAssertEqual(certificate.name, "Mac Installer Distribution: Hello")
+                    XCTAssertEqual(certificate.platform, BundleIdPlatform.macOS)
+                    XCTAssertEqual(certificate.content, "MIIFpDCCBIygAwIBAgIIbgb/7NS42MgwDQ")
+                })
```
```suggestion
            case .success(let certificate):
                XCTAssertEqual(certificate.name, "Mac Installer Distribution: Hello")
                XCTAssertEqual(certificate.platform, BundleIdPlatform.macOS)
                XCTAssertEqual(certificate.content, "MIIFpDCCBIygAwIBAgIIbgb/7NS42MgwDQ")
```

adamjcampbell commented on Tests/appstoreconnect-cliTests/Operations/CreateCertificateOperationTests.swift:
```diff
@@ -0,0 +1,115 @@
+        certificateType: .iOSDevelopment,
+        csrContent: "")
+
+    func testCreateCertificate_success() {
```
🤔 when the test outputs the result we get a little bit of redundancy here:
e.g 
`Test Case '-[appstoreconnect_cliTests.CreateCertificateOperationTests testCreateCertificate_success]' passed (0.099 seconds).`

Maybe we should shorten the prefixes to just the function we are testing:
```suggestion
    func testExecute_success() {
```
